src/main/java/net/minecraftforge/fluids/FluidAttributes.java

```diff
@@ -338,22 +353,31 @@ public SoundEvent getEmptySound()
     public boolean isGaseous(IEnviromentBlockReader world, BlockPos pos){ return isGaseous(); }
     public Rarity getRarity(IEnviromentBlockReader world, BlockPos pos){ return getRarity(); }
     public int getColor(IEnviromentBlockReader world, BlockPos pos){ return getColor(); }
+    public int getFuelTime(World world, BlockPos pos){ return getFuelTime(); }
+    public boolean isFuel(World world, BlockPos pos){ return isFuel(); }
     public ResourceLocation getStill(IEnviromentBlockReader world, BlockPos pos) { return getStillTexture(); }
     public ResourceLocation getFlowing(IEnviromentBlockReader world, BlockPos pos) { return getFlowingTexture(); }
     public SoundEvent getFillSound(IEnviromentBlockReader world, BlockPos pos) { return getFillSound(); }
     public SoundEvent getEmptySound(IEnviromentBlockReader world, BlockPos pos) { return getEmptySound(); }
 
-    public static Builder builder(String name, ResourceLocation stillTexture, ResourceLocation flowingTexture) {
-        return new Builder(name, stillTexture, flowingTexture);
+    public static Builder builder(Fluid owner, ResourceLocation stillTexture, ResourceLocation flowingTexture) {
+        return new Builder(owner, stillTexture, flowingTexture);
+    }
+
+    public Stream<ResourceLocation> getTextures()
+    {
+        if (overlayTexture != null)
+            return Stream.of(stillTexture, flowingTexture, overlayTexture);
+        return Stream.of(stillTexture, flowingTexture);
     }
 
     public static class Builder
     {
-        private final String name;
+        private final Fluid owner;
```

Comment:
Why does this need to be the fluid instance? I can see this being a chicken-egg style thing. 
Have you tried using a registry delegate or something that isn't the Fluid directly.
A delegate would silently support registry replacements. But doesn't solve the chicken/egg thing off hand.
All you seem to use it for is the name, what was the logic behind this change?

Reply:
My thinking was that it may be useful for the people subclassing FluidAttributes, but on second thought, all methods that are worth overriding either have a FluidStack or a FluidState, meaning both have a .getFluid() method available, so the field could be simplified to a ResourceLocation.

Reply:
Removed the pointless field. The builder function still takes a Fluid, for convenience, but it's only used to get the registry name from.